### PR TITLE
Ghost ass slapping has arrived

### PR DESCRIPTION
**Pull request by PurpleFenn**
# About The Pull Request
With the GHOST_INTERACTION config flag, ghosts can now slap the ass of players who have the ass slap pref enabled! Neat!

## Why It's Good For The Game
Why not? It's able to be disabled pretty easily via a config flag.

## A Port?
Not at all

## Pre-Merge Checklist

- [x] You tested this on a local server.
- [ ] This code did not runtime during testing.
- [x] You documented all of your changes.

## Changelog

:cl:
add: The ability to slap the ass of physical beings as a ghost. Spooky ☠☠
/:cl:

### CODE REVIEW

**Comment by PurpleFenn** [comment]
Leaving runtime unticked because it's not fully tested. Ass slapping works fine but there might still be cases in which runtimes occur since I specifically used the syntax that allows runtimes to happen instead of compilation error to avoid having to deal with some bs later

File: modular_splurt/code/_onclick/observer.dm [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+		return
+	playsound(src.loc, 'sound/weapons/slap.ogg', 50, 1, -1)
+	if(HAS_TRAIT(src, TRAIT_STEEL_ASS))
+		to_chat(src, "<span class='danger'>You feel something bounce off your steely asscheeks, but nothing is there...</span>")
+		to_chat(user, "<span class='danger'>You slap \The [src]'s ass, but your ethereal hand bounces right off!</span>")
+		playsound(src.loc, 'sound/weapons/tap.ogg', 50, 1, -1)
+		return
+	src:adjust_arousal(20, "masochism", maso=TRUE)	// Honestly fuck you, just runtime
+	if(ishuman(src) && HAS_TRAIT(src, TRAIT_MASO) && src.has_dna() && prob(10))
+		src:mob_climax(forced_climax=TRUE, cause="masochism")
```
Comment (BongaTheProto) [inline modular_splurt/code/_onclick/observer.dm:28]:
```suggestion
	if(istype(H))
		H.adjust_arousal(20, "masochism", maso=TRUE)
		if(HAS_TRAIT(H, TRAIT_MASO) && H.has_dna() && prob(10))
			H.mob_climax(forced_climax=TRUE, cause="masochism")
```

Comment (BongaTheProto) [inline modular_splurt/code/_onclick/observer.dm:28]:
```suggestion
	if(istype(H))
		H.adjust_arousal(20, "masochism", maso=TRUE)
		if(HAS_TRAIT(H, TRAIT_MASO) && H.has_dna() && prob(10))
			H.mob_climax(forced_climax=TRUE, cause="masochism")
```

File: modular_splurt/code/_onclick/observer.dm [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+			attack_ai(user)
+		else if(user.client.prefs.inquisitive_ghost)
+			user.examinate(src)
+		else if(CONFIG_GET(flag/ghost_interaction) && istype(src, /mob/living))	// This is what's new
+			var/mob/living/H = src
+			H.do_ass_slap(user)
+
+// I'm putting this here because honestly i'm too lazy to find the thing i need
+/mob/living/proc/do_ass_slap(mob/dead/observer/user)
+	var/mob/living/carbon/H = src
```
Comment (BongaTheProto) [inline modular_splurt/code/_onclick/observer.dm:16]:
```suggestion
	var/mob/living/carbon/human/H = src
```

Comment (BongaTheProto) [inline modular_splurt/code/_onclick/observer.dm:16]:
```suggestion
	var/mob/living/carbon/human/H = src
```

**Review by BongaTheProto — CHANGES_REQUESTED** [review]

**Comment by BongaTheProto** [comment]
I think this will probably be better

**Review by BongaTheProto — APPROVED** [review]

**Pull request merged**